<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8"/><title>5. Distal regulatory activity</title><style>
/* cspell:disable-file */
/* webkit printing magic: print all background colors */
html {
	-webkit-print-color-adjust: exact;
}
* {
	box-sizing: border-box;
	-webkit-print-color-adjust: exact;
}

html,
body {
	margin: 0;
	padding: 0;
}
@media only screen {
	body {
		margin: 2em auto;
		max-width: 900px;
		color: rgb(55, 53, 47);
	}
}

body {
	line-height: 1.5;
	white-space: pre-wrap;
}

a,
a.visited {
	color: inherit;
	text-decoration: underline;
}

.pdf-relative-link-path {
	font-size: 80%;
	color: #444;
}

h1,
h2,
h3 {
	letter-spacing: -0.01em;
	line-height: 1.2;
	font-weight: 600;
	margin-bottom: 0;
}

.page-title {
	font-size: 2.5rem;
	font-weight: 700;
	margin-top: 0;
	margin-bottom: 0.75em;
}

h1 {
	font-size: 1.875rem;
	margin-top: 1.875rem;
}

h2 {
	font-size: 1.5rem;
	margin-top: 1.5rem;
}

h3 {
	font-size: 1.25rem;
	margin-top: 1.25rem;
}

.source {
	border: 1px solid #ddd;
	border-radius: 3px;
	padding: 1.5em;
	word-break: break-all;
}

.callout {
	border-radius: 3px;
	padding: 1rem;
}

figure {
	margin: 1.25em 0;
	page-break-inside: avoid;
}

figcaption {
	opacity: 0.5;
	font-size: 85%;
	margin-top: 0.5em;
}

mark {
	background-color: transparent;
}

.indented {
	padding-left: 1.5em;
}

hr {
	background: transparent;
	display: block;
	width: 100%;
	height: 1px;
	visibility: visible;
	border: none;
	border-bottom: 1px solid rgba(55, 53, 47, 0.09);
}

img {
	max-width: 100%;
}

@media only print {
	img {
		max-height: 100vh;
		object-fit: contain;
	}
}

@page {
	margin: 1in;
}

.collection-content {
	font-size: 0.875rem;
}

.column-list {
	display: flex;
	justify-content: space-between;
}

.column {
	padding: 0 1em;
}

.column:first-child {
	padding-left: 0;
}

.column:last-child {
	padding-right: 0;
}

.table_of_contents-item {
	display: block;
	font-size: 0.875rem;
	line-height: 1.3;
	padding: 0.125rem;
}

.table_of_contents-indent-1 {
	margin-left: 1.5rem;
}

.table_of_contents-indent-2 {
	margin-left: 3rem;
}

.table_of_contents-indent-3 {
	margin-left: 4.5rem;
}

.table_of_contents-link {
	text-decoration: none;
	opacity: 0.7;
	border-bottom: 1px solid rgba(55, 53, 47, 0.18);
}

table,
th,
td {
	border: 1px solid rgba(55, 53, 47, 0.09);
	border-collapse: collapse;
}

table {
	border-left: none;
	border-right: none;
}

th,
td {
	font-weight: normal;
	padding: 0.25em 0.5em;
	line-height: 1.5;
	min-height: 1.5em;
	text-align: left;
}

th {
	color: rgba(55, 53, 47, 0.6);
}

ol,
ul {
	margin: 0;
	margin-block-start: 0.6em;
	margin-block-end: 0.6em;
}

li > ol:first-child,
li > ul:first-child {
	margin-block-start: 0.6em;
}

ul > li {
	list-style: disc;
}

ul.to-do-list {
	padding-inline-start: 0;
}

ul.to-do-list > li {
	list-style: none;
}

.to-do-children-checked {
	text-decoration: line-through;
	opacity: 0.375;
}

ul.toggle > li {
	list-style: none;
}

ul {
	padding-inline-start: 1.7em;
}

ul > li {
	padding-left: 0.1em;
}

ol {
	padding-inline-start: 1.6em;
}

ol > li {
	padding-left: 0.2em;
}

.mono ol {
	padding-inline-start: 2em;
}

.mono ol > li {
	text-indent: -0.4em;
}

.toggle {
	padding-inline-start: 0em;
	list-style-type: none;
}

/* Indent toggle children */
.toggle > li > details {
	padding-left: 1.7em;
}

.toggle > li > details > summary {
	margin-left: -1.1em;
}

.selected-value {
	display: inline-block;
	padding: 0 0.5em;
	background: rgba(206, 205, 202, 0.5);
	border-radius: 3px;
	margin-right: 0.5em;
	margin-top: 0.3em;
	margin-bottom: 0.3em;
	white-space: nowrap;
}

.collection-title {
	display: inline-block;
	margin-right: 1em;
}

.page-description {
    margin-bottom: 2em;
}

.simple-table {
	margin-top: 1em;
	font-size: 0.875rem;
	empty-cells: show;
}
.simple-table td {
	height: 29px;
	min-width: 120px;
}

.simple-table th {
	height: 29px;
	min-width: 120px;
}

.simple-table-header-color {
	background: rgb(247, 246, 243);
	color: black;
}
.simple-table-header {
	font-weight: 500;
}

time {
	opacity: 0.5;
}

.icon {
	display: inline-block;
	max-width: 1.2em;
	max-height: 1.2em;
	text-decoration: none;
	vertical-align: text-bottom;
	margin-right: 0.5em;
}

img.icon {
	border-radius: 3px;
}

.user-icon {
	width: 1.5em;
	height: 1.5em;
	border-radius: 100%;
	margin-right: 0.5rem;
}

.user-icon-inner {
	font-size: 0.8em;
}

.text-icon {
	border: 1px solid #000;
	text-align: center;
}

.page-cover-image {
	display: block;
	object-fit: cover;
	width: 100%;
	max-height: 30vh;
}

.page-header-icon {
	font-size: 3rem;
	margin-bottom: 1rem;
}

.page-header-icon-with-cover {
	margin-top: -0.72em;
	margin-left: 0.07em;
}

.page-header-icon img {
	border-radius: 3px;
}

.link-to-page {
	margin: 1em 0;
	padding: 0;
	border: none;
	font-weight: 500;
}

p > .user {
	opacity: 0.5;
}

td > .user,
td > time {
	white-space: nowrap;
}

input[type="checkbox"] {
	transform: scale(1.5);
	margin-right: 0.6em;
	vertical-align: middle;
}

p {
	margin-top: 0.5em;
	margin-bottom: 0.5em;
}

.image {
	border: none;
	margin: 1.5em 0;
	padding: 0;
	border-radius: 0;
	text-align: center;
}

.code,
code {
	background: rgba(135, 131, 120, 0.15);
	border-radius: 3px;
	padding: 0.2em 0.4em;
	border-radius: 3px;
	font-size: 85%;
	tab-size: 2;
}

code {
	color: #eb5757;
}

.code {
	padding: 1.5em 1em;
}

.code-wrap {
	white-space: pre-wrap;
	word-break: break-all;
}

.code > code {
	background: none;
	padding: 0;
	font-size: 100%;
	color: inherit;
}

blockquote {
	font-size: 1.25em;
	margin: 1em 0;
	padding-left: 1em;
	border-left: 3px solid rgb(55, 53, 47);
}

.bookmark {
	text-decoration: none;
	max-height: 8em;
	padding: 0;
	display: flex;
	width: 100%;
	align-items: stretch;
}

.bookmark-title {
	font-size: 0.85em;
	overflow: hidden;
	text-overflow: ellipsis;
	height: 1.75em;
	white-space: nowrap;
}

.bookmark-text {
	display: flex;
	flex-direction: column;
}

.bookmark-info {
	flex: 4 1 180px;
	padding: 12px 14px 14px;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
}

.bookmark-image {
	width: 33%;
	flex: 1 1 180px;
	display: block;
	position: relative;
	object-fit: cover;
	border-radius: 1px;
}

.bookmark-description {
	color: rgba(55, 53, 47, 0.6);
	font-size: 0.75em;
	overflow: hidden;
	max-height: 4.5em;
	word-break: break-word;
}

.bookmark-href {
	font-size: 0.75em;
	margin-top: 0.25em;
}

.sans { font-family: ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol"; }
.code { font-family: "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace; }
.serif { font-family: Lyon-Text, Georgia, ui-serif, serif; }
.mono { font-family: iawriter-mono, Nitti, Menlo, Courier, monospace; }
.pdf .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK JP'; }
.pdf:lang(zh-CN) .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK SC'; }
.pdf:lang(zh-TW) .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK TC'; }
.pdf:lang(ko-KR) .sans { font-family: Inter, ui-sans-serif, -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, "Apple Color Emoji", Arial, sans-serif, "Segoe UI Emoji", "Segoe UI Symbol", 'Twemoji', 'Noto Color Emoji', 'Noto Sans CJK KR'; }
.pdf .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK JP'; }
.pdf:lang(zh-CN) .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK SC'; }
.pdf:lang(zh-TW) .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK TC'; }
.pdf:lang(ko-KR) .code { font-family: Source Code Pro, "SFMono-Regular", Menlo, Consolas, "PT Mono", "Liberation Mono", Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK KR'; }
.pdf .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK JP'; }
.pdf:lang(zh-CN) .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK SC'; }
.pdf:lang(zh-TW) .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK TC'; }
.pdf:lang(ko-KR) .serif { font-family: PT Serif, Lyon-Text, Georgia, ui-serif, serif, 'Twemoji', 'Noto Color Emoji', 'Noto Serif CJK KR'; }
.pdf .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK JP'; }
.pdf:lang(zh-CN) .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK SC'; }
.pdf:lang(zh-TW) .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK TC'; }
.pdf:lang(ko-KR) .mono { font-family: PT Mono, iawriter-mono, Nitti, Menlo, Courier, monospace, 'Twemoji', 'Noto Color Emoji', 'Noto Sans Mono CJK KR'; }
.highlight-default {
	color: rgba(55, 53, 47, 1);
}
.highlight-gray {
	color: rgba(120, 119, 116, 1);
	fill: rgba(120, 119, 116, 1);
}
.highlight-brown {
	color: rgba(159, 107, 83, 1);
	fill: rgba(159, 107, 83, 1);
}
.highlight-orange {
	color: rgba(217, 115, 13, 1);
	fill: rgba(217, 115, 13, 1);
}
.highlight-yellow {
	color: rgba(203, 145, 47, 1);
	fill: rgba(203, 145, 47, 1);
}
.highlight-teal {
	color: rgba(68, 131, 97, 1);
	fill: rgba(68, 131, 97, 1);
}
.highlight-blue {
	color: rgba(51, 126, 169, 1);
	fill: rgba(51, 126, 169, 1);
}
.highlight-purple {
	color: rgba(144, 101, 176, 1);
	fill: rgba(144, 101, 176, 1);
}
.highlight-pink {
	color: rgba(193, 76, 138, 1);
	fill: rgba(193, 76, 138, 1);
}
.highlight-red {
	color: rgba(212, 76, 71, 1);
	fill: rgba(212, 76, 71, 1);
}
.highlight-gray_background {
	background: rgba(241, 241, 239, 1);
}
.highlight-brown_background {
	background: rgba(244, 238, 238, 1);
}
.highlight-orange_background {
	background: rgba(251, 236, 221, 1);
}
.highlight-yellow_background {
	background: rgba(251, 243, 219, 1);
}
.highlight-teal_background {
	background: rgba(237, 243, 236, 1);
}
.highlight-blue_background {
	background: rgba(231, 243, 248, 1);
}
.highlight-purple_background {
	background: rgba(244, 240, 247, 0.8);
}
.highlight-pink_background {
	background: rgba(249, 238, 243, 0.8);
}
.highlight-red_background {
	background: rgba(253, 235, 236, 1);
}
.block-color-default {
	color: inherit;
	fill: inherit;
}
.block-color-gray {
	color: rgba(120, 119, 116, 1);
	fill: rgba(120, 119, 116, 1);
}
.block-color-brown {
	color: rgba(159, 107, 83, 1);
	fill: rgba(159, 107, 83, 1);
}
.block-color-orange {
	color: rgba(217, 115, 13, 1);
	fill: rgba(217, 115, 13, 1);
}
.block-color-yellow {
	color: rgba(203, 145, 47, 1);
	fill: rgba(203, 145, 47, 1);
}
.block-color-teal {
	color: rgba(68, 131, 97, 1);
	fill: rgba(68, 131, 97, 1);
}
.block-color-blue {
	color: rgba(51, 126, 169, 1);
	fill: rgba(51, 126, 169, 1);
}
.block-color-purple {
	color: rgba(144, 101, 176, 1);
	fill: rgba(144, 101, 176, 1);
}
.block-color-pink {
	color: rgba(193, 76, 138, 1);
	fill: rgba(193, 76, 138, 1);
}
.block-color-red {
	color: rgba(212, 76, 71, 1);
	fill: rgba(212, 76, 71, 1);
}
.block-color-gray_background {
	background: rgba(241, 241, 239, 1);
}
.block-color-brown_background {
	background: rgba(244, 238, 238, 1);
}
.block-color-orange_background {
	background: rgba(251, 236, 221, 1);
}
.block-color-yellow_background {
	background: rgba(251, 243, 219, 1);
}
.block-color-teal_background {
	background: rgba(237, 243, 236, 1);
}
.block-color-blue_background {
	background: rgba(231, 243, 248, 1);
}
.block-color-purple_background {
	background: rgba(244, 240, 247, 0.8);
}
.block-color-pink_background {
	background: rgba(249, 238, 243, 0.8);
}
.block-color-red_background {
	background: rgba(253, 235, 236, 1);
}
.select-value-color-uiBlue { background-color: rgba(35, 131, 226, .07); }
.select-value-color-pink { background-color: rgba(245, 224, 233, 1); }
.select-value-color-purple { background-color: rgba(232, 222, 238, 1); }
.select-value-color-green { background-color: rgba(219, 237, 219, 1); }
.select-value-color-gray { background-color: rgba(227, 226, 224, 1); }
.select-value-color-translucentGray { background-color: rgba(255, 255, 255, 0.0375); }
.select-value-color-orange { background-color: rgba(250, 222, 201, 1); }
.select-value-color-brown { background-color: rgba(238, 224, 218, 1); }
.select-value-color-red { background-color: rgba(255, 226, 221, 1); }
.select-value-color-yellow { background-color: rgba(253, 236, 200, 1); }
.select-value-color-blue { background-color: rgba(211, 229, 239, 1); }
.select-value-color-pageGlass { background-color: undefined; }
.select-value-color-washGlass { background-color: undefined; }

.checkbox {
	display: inline-flex;
	vertical-align: text-bottom;
	width: 16;
	height: 16;
	background-size: 16px;
	margin-left: 2px;
	margin-right: 5px;
}

.checkbox-on {
	background-image: url("data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%2216%22%20height%3D%2216%22%20viewBox%3D%220%200%2016%2016%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%3Crect%20width%3D%2216%22%20height%3D%2216%22%20fill%3D%22%2358A9D7%22%2F%3E%0A%3Cpath%20d%3D%22M6.71429%2012.2852L14%204.9995L12.7143%203.71436L6.71429%209.71378L3.28571%206.2831L2%207.57092L6.71429%2012.2852Z%22%20fill%3D%22white%22%2F%3E%0A%3C%2Fsvg%3E");
}

.checkbox-off {
	background-image: url("data:image/svg+xml;charset=UTF-8,%3Csvg%20width%3D%2216%22%20height%3D%2216%22%20viewBox%3D%220%200%2016%2016%22%20fill%3D%22none%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%0A%3Crect%20x%3D%220.75%22%20y%3D%220.75%22%20width%3D%2214.5%22%20height%3D%2214.5%22%20fill%3D%22white%22%20stroke%3D%22%2336352F%22%20stroke-width%3D%221.5%22%2F%3E%0A%3C%2Fsvg%3E");
}
	
</style></head><body><article id="ec3ddb36-536c-4830-abc1-c4af5d3d1580" class="page mono"><header><h1 class="page-title">5. Distal regulatory activity</h1><p class="page-description"></p></header><div class="page-body"><p id="76f455fe-bc25-4468-b6d8-da3881e38c6b" class="">
</p><p id="21091ffd-d038-41f1-8667-92d78a5df538" class="">Inside <code>epigenomics_uvic</code>, run</p><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="d0d0ba96-789e-4f17-9738-19f66dd3747d" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">sudo docker run -v $PWD:$PWD -w $PWD --rm -it dgarrimar/epigenomics_course</code></pre><details open=""><summary style="font-weight:600;font-size:1.875em;line-height:1.3;margin:0">Tasks:</summary><div class="indented"><ul id="17a421bb-a9de-46e9-b3e9-392c55feb662" class="toggle"><li><details open=""><summary>Create a folder regulatory_elements inside <code>epigenomics_uvic</code>. This will be the folder where you store all your subsequent results:</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="6a9f8d8b-6fdf-4da7-ac90-b8b9496b7ce4" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">mkdir regulatory_elements
mkdir regulatory_elements/data
mkdir regulatory_elements/data/tsv
mkdir regulatory_elements/data/bigBed.files
mkdir regulatory_elements/data/bed.files
mkdir regulatory_elements/analyses
mkdir regulatory_elements/analyses/peaks.analyses</code></pre></details></li></ul><ul id="b5063c12-8425-41b0-aab7-2c41d2e167f4" class="toggle"><li><details open=""><summary>Select the open regions that overlap peaks of H3K27ac and H3K4me1 in the corresponding tissue. You will get a list of candidate distal regulatory elements for each tissue:</summary><ol type="1" id="8f8d30a8-19d8-404c-b383-00a44972f6cf" class="numbered-list" start="1"><li>Download peak calling files for H3K27ac and H3K4me1:<ul id="365f1ddf-a9c8-4fd0-aad7-42d2564c9fc0" class="toggle"><li><details open=""><summary>For H3K27ac:</summary><ul id="12d91476-c3eb-4c51-b600-b90f27adb910" class="toggle"><li><details open=""><summary>Get the ids of the bigBed files:</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="eaa1617b-7a2a-4cfe-8334-274bc6f87f71" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">grep -F H3K27ac ../Chip-seq/metadata.tsv |\
grep -F &quot;bigBed_narrowPeak&quot; |\
grep -F &quot;pseudoreplicated_peaks&quot; |\
grep -F &quot;GRCh38&quot; |\
awk &#x27;BEGIN{FS=OFS=&quot;\t&quot;}{print $1, $11, $23}&#x27; |\
sort -k2,2 -k1,1r |\
sort -k2,2 -u &gt; regulatory_elements/data/bigBed.files/bigBed.H3K27ac.ids.txt</code></pre></details></li></ul><ul id="f5afcfc9-cf76-41b0-8645-f32470813aa2" class="toggle"><li><details open=""><summary>Download the bigBed files:</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="8fdd1ee8-c5f0-49fa-9e71-8af9e716913a" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">cut -f1 regulatory_elements/data/bigBed.files/bigBed.H3K27ac.ids.txt |\
while read filename; do
wget -P regulatory_elements/data/bigBed.files &quot;https://www.encodeproject.org/files/$filename/@@download/$filename.bigBed&quot;

done</code></pre></details></li></ul><ul id="9b48bb2d-e532-4518-b3ad-f2a7e18b75c3" class="toggle"><li><details open=""><summary>Convert bigBed to bed files:</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="b789768e-e51f-4a28-9e4a-4f570902ee67" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">cut -f1 regulatory_elements/data/bigBed.files/bigBed.H3K27ac.ids.txt |\
while read filename; do
bigBedToBed regulatory_elements/data/bigBed.files/&quot;$filename&quot;.bigBed regulatory_elements/data/bed.files/&quot;$filename&quot;.bed

done</code></pre></details></li></ul><ul id="f3f4c800-7516-471c-afc2-b30bd43cc781" class="toggle"><li><details open=""><summary>Making sure that my md5sum values coincide with one provided by ENCODE</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="35207e76-5596-478f-8f18-5953d89e7553" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all"># Define file type
file_type=&quot;bigBed&quot;

# Loop over each file of the defined type
for file_type in bigBed; do
    # Define the input file path
    input_file=&quot;regulatory_elements/data/bigBed.files/${file_type}.H3K27ac.ids.txt&quot;
    md5sum_file=&quot;regulatory_elements/data/${file_type}.files/md5sum_H3K27ac.txt&quot;

    # Check if the input file exists
    if [ -f &quot;$input_file&quot; ]; then
        echo &quot;Processing $input_file...&quot;

        # Run the selectRows.sh script and store the result in md5sum.txt
        if ../bin/selectRows.sh &lt;(cut -f1 &quot;$input_file&quot;) ../ChIP-seq/metadata.tsv | cut -f1,46 &gt; &quot;$md5sum_file&quot;; then
            echo &quot;MD5 checksums are stored in $md5sum_file.&quot;
        else
            echo &quot;An error occurred while running selectRows.sh script.&quot;
            continue
        fi
    else
        echo &quot;Input file $input_file does not exist. Skipping this file.&quot;
        continue
    fi

    # Check if the md5sum file exists
    if [ -f &quot;$md5sum_file&quot; ]; then
        echo &quot;Starting the verification of $md5sum_file...&quot;

        # Read each line from the md5sum file
        while read -r filename original_md5sum; do
            # Check if the file exists
            if [ -f &quot;regulatory_elements/data/${file_type}.files/${filename}.${file_type}&quot; ]; then
                # Calculate the MD5 checksum of the file and compare it with the original
                calculated_md5sum=$(md5sum &quot;regulatory_elements/data/${file_type}.files/${filename}.${file_type}&quot; | cut -d &#x27; &#x27; -f 1)

                # Print the filename, original MD5 checksum, and calculated MD5 checksum
                echo -e &quot;$filename\t$original_md5sum\t$calculated_md5sum&quot;
            else
                echo &quot;File regulatory_elements/data/${file_type}.files/${filename}.${file_type} does not exist. Skipping this file.&quot;
            fi
        done &lt; &quot;$md5sum_file&quot; &gt; tmp &amp;&amp; mv tmp &quot;$md5sum_file&quot;

        echo &quot;The verification of $md5sum_file is completed. The results are stored in $md5sum_file.&quot;
    else
        echo &quot;The MD5 checksum file $md5sum_file does not exist. Skipping this file.&quot;
    fi
done</code></pre></details></li></ul><ul id="0c18edeb-814e-42d0-a76f-8bfeaa2e0697" class="toggle"><li><details open=""><summary>make sure there are no files for which original and computed MD5 hashes differ</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="edc798aa-aff2-4e04-a623-d895f004e08a" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">awk &#x27;$2!=$3&#x27; regulatory_elements/data/&quot;$file_type&quot;.files/md5sum_H3K27ac.txt

done</code></pre></details></li></ul></details></li></ul><ul id="6783df3a-b97e-4f17-8921-8fa23cefe6f8" class="toggle"><li><details open=""><summary>For H3K4me1:</summary><ul id="634ec515-28dd-4d2e-9889-8f8b00da5d7d" class="toggle"><li><details open=""><summary>Get the ids of the bigBed files:</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="872e55b0-02ca-467f-b609-60dc41785a50" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">grep -F H3K4me1 ../Chip-seq/metadata.tsv |\
grep -F &quot;bigBed_narrowPeak&quot; |\
grep -F &quot;pseudoreplicated_peaks&quot; |\
grep -F &quot;GRCh38&quot; |\
awk &#x27;BEGIN{FS=OFS=&quot;\t&quot;}{print $1, $11, $23}&#x27; |\
sort -k2,2 -k1,1r |\
sort -k2,2 -u &gt; regulatory_elements/data/bigBed.files/bigBed.H3K4me1.ids.txt</code></pre></details></li></ul><ul id="4101c701-6390-4765-8d47-7b3f4ed0710a" class="toggle"><li><details open=""><summary>Download the bigBed files</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="acfcce77-e195-4375-aa2a-e688b54f3a2d" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">cut -f1 regulatory_elements/data/bigBed.files/bigBed.H3K4me1.ids.txt |\
while read filename; do
wget -P regulatory_elements/data/bigBed.files &quot;https://www.encodeproject.org/files/$filename/@@download/$filename.bigBed&quot;

done</code></pre></details></li></ul><ul id="007daf90-59da-4bdf-a0d5-0a381dda9df3" class="toggle"><li><details open=""><summary>Convert bigBed to bed files</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="4e4d761c-5719-41a3-85d3-4fea25d507c6" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">cut -f1 regulatory_elements/data/bigBed.files/bigBed.H3K4me1.ids.txt |\
while read filename; do
bigBedToBed regulatory_elements/data/bigBed.files/&quot;$filename&quot;.bigBed regulatory_elements/data/bed.files/&quot;$filename&quot;.bed

done</code></pre></details></li></ul><ul id="4361bdc2-068a-4f55-9ee5-0ce8e7bb68b6" class="toggle"><li><details open=""><summary>Making sure that my md5sum values coincide with one provided by ENCODE</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="d351c166-5f63-4d1a-bf5b-4fe077d53a04" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all"># Define file type
file_type=&quot;bigBed&quot;

# Loop over each file of the defined type
for file_type in bigBed; do
    # Define the input file path
    input_file=&quot;regulatory_elements/data/bigBed.files/${file_type}.H3K4me1.ids.txt&quot;
    md5sum_file=&quot;regulatory_elements/data/${file_type}.files/md5sum_H3K4me1.txt&quot;

    # Check if the input file exists
    if [ -f &quot;$input_file&quot; ]; then
        echo &quot;Processing $input_file...&quot;

        # Run the selectRows.sh script and store the result in md5sum.txt
        if ../bin/selectRows.sh &lt;(cut -f1 &quot;$input_file&quot;) ../ChIP-seq/metadata.tsv | cut -f1,46 &gt; &quot;$md5sum_file&quot;; then
            echo &quot;MD5 checksums are stored in $md5sum_file.&quot;
        else
            echo &quot;An error occurred while running selectRows.sh script.&quot;
            continue
        fi
    else
        echo &quot;Input file $input_file does not exist. Skipping this file.&quot;
        continue
    fi

    # Check if the md5sum file exists
    if [ -f &quot;$md5sum_file&quot; ]; then
        echo &quot;Starting the verification of $md5sum_file...&quot;

        # Read each line from the md5sum file
        while read -r filename original_md5sum; do
            # Check if the file exists
            if [ -f &quot;regulatory_elements/data/${file_type}.files/${filename}.${file_type}&quot; ]; then
                # Calculate the MD5 checksum of the file and compare it with the original
                calculated_md5sum=$(md5sum &quot;regulatory_elements/data/${file_type}.files/${filename}.${file_type}&quot; | cut -d &#x27; &#x27; -f 1)

                # Print the filename, original MD5 checksum, and calculated MD5 checksum
                echo -e &quot;$filename\t$original_md5sum\t$calculated_md5sum&quot;
            else
                echo &quot;File regulatory_elements/data/${file_type}.files/${filename}.${file_type} does not exist. Skipping this file.&quot;
            fi
        done &lt; &quot;$md5sum_file&quot; &gt; tmp &amp;&amp; mv tmp &quot;$md5sum_file&quot;

        echo &quot;The verification of $md5sum_file is completed. The results are stored in $md5sum_file.&quot;
    else
        echo &quot;The MD5 checksum file $md5sum_file does not exist. Skipping this file.&quot;
    fi

done</code></pre></details></li></ul><ul id="db766f20-4e7c-4b01-8d20-d8e81744656b" class="toggle"><li><details open=""><summary>make sure there are no files for which original and computed MD5 hashes differ</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="a631b483-0b4e-4a3c-aa54-e32fd06db731" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">awk &#x27;$2!=$3&#x27; regulatory_elements/data/&quot;$file_type&quot;.files/md5sum_H3K4me1.txt

done</code></pre></details></li></ul></details></li></ul></li></ol><ol type="1" id="ab7bd551-2ad7-42e3-9b81-3481ed0e18b8" class="numbered-list" start="2"><li>Select open regions overlapping with H3K27ac and H3K4me1 in each tissue:<ul id="420a89c5-1ff5-4da5-b825-fdb88a5c5be0" class="toggle"><li><details open=""><summary>The Bash script intersects ATAC-seq peaks with H3K27ac and H3K4me1 marks for each tissue, then finds the common peaks between H3K27ac and H3K4me1, and finally counts the number of peaks in each tissue-specific file and in all files combined.</summary><figure class="block-color-gray_background callout" style="white-space:pre-wrap;display:flex" id="485686de-1d4e-479d-ba85-c3e346519a39"><div style="font-size:1.5em"><span class="icon">💡</span></div><div style="width:100%">The <code>peak_analysis.sh</code> script is located in the <code>bin</code> directory under the <code>ATAC-seq</code> project, which is a part of the <code>epigenomics_uvic</code> study.</div></figure><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="1e3b5bf2-9805-450a-8970-3b809295908a" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">#!/bin/bash

# Define the list of tissues
tissues=(&quot;sigmoid_colon&quot; &quot;stomach&quot;)

# Print a header
echo &quot;==================== Peak Analysis ====================&quot;

# Loop over each tissue
for tissue in &quot;${tissues[@]}&quot;; do
    echo &quot;-----------------------------------------------------&quot;
    echo &quot;Processing $tissue...&quot;

    # Intersect ATAC-seq peaks with H3K27ac mark
    echo &quot;Intersecting ATAC-seq peaks with H3K27ac mark for $tissue...&quot;
    bedtools intersect -a analyses/peaks.analysis/peaks.not.body.&quot;$tissue&quot;.bed -b regulatory_elements/data/bed.files/$(awk -v tissue=&quot;$tissue&quot; &#x27;$2 == tissue {print $1}&#x27; regulatory_elements/data/bigBed.files/bigBed.H3K27ac.ids.txt).bed -u &gt; regulatory_elements/analyses/peaks.analyses/peaks.regulatory.H3K27ac.&quot;$tissue&quot;.bed
    
    echo &quot;Done with H3K27ac intersection for $tissue.&quot;

    # Intersect ATAC-seq peaks with H3K4me1 mark
    echo &quot;Intersecting ATAC-seq peaks with H3K4me1 mark for $tissue...&quot;
    bedtools intersect -a analyses/peaks.analysis/peaks.not.body.&quot;$tissue&quot;.bed -b regulatory_elements/data/bed.files/$(awk -v tissue=&quot;$tissue&quot; &#x27;$2 == tissue {print $1}&#x27; regulatory_elements/data/bigBed.files/bigBed.H3K4me1.ids.txt).bed -u &gt; regulatory_elements/analyses/peaks.analyses/peaks.regulatory.H3K4me1.&quot;$tissue&quot;.bed

    echo &quot;Done with H3K4me1 intersection for $tissue.&quot;

    # Intersect H3K27ac and H3K4me1 peaks
    echo &quot;Intersecting H3K27ac and H3K4me1 peaks for $tissue...&quot;
    bedtools intersect -a regulatory_elements/analyses/peaks.analyses/peaks.regulatory.H3K4me1.&quot;$tissue&quot;.bed -b regulatory_elements/analyses/peaks.analyses/peaks.regulatory.H3K27ac.&quot;$tissue&quot;.bed -u &gt; regulatory_elements/analyses/peaks.analyses/peaks.regulatory.H3K4me1.AND.H3K27ac.&quot;$tissue&quot;.bed

    echo &quot;Done with H3K27ac and H3K4me1 intersection for $tissue.&quot;

    # Count the number of lines (peaks) in each file
    num_peaks=$(wc -l &lt; regulatory_elements/analyses/peaks.analyses/peaks.regulatory.H3K4me1.AND.H3K27ac.&quot;$tissue&quot;.bed)
    echo &quot;The file for $tissue contains $num_peaks peaks.&quot;
done

# Count the total number of lines (peaks) in all files
total_peaks=$(cat regulatory_elements/analyses/peaks.analyses/peaks.regulatory.H3K4me1.AND.H3K27ac.*.bed | wc -l)
echo &quot;-----------------------------------------------------&quot;
echo &quot;The total number of peaks in all files is $total_peaks.&quot;

echo &quot;==================== All done! ====================&quot;</code></pre></details></li></ul><ul id="af8115c3-2f57-4fdc-af69-0850e50fa661" class="toggle"><li><details open=""><summary>Results:</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="e87c028a-2faf-42f2-a0e3-a650ea6b304b" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">==================== Peak Analysis ====================
-----------------------------------------------------
Processing sigmoid_colon...
Intersecting ATAC-seq peaks with H3K27ac mark for sigmoid_colon...
Done with H3K27ac intersection for sigmoid_colon.
Intersecting ATAC-seq peaks with H3K4me1 mark for sigmoid_colon...
Done with H3K4me1 intersection for sigmoid_colon.
Intersecting H3K27ac and H3K4me1 peaks for sigmoid_colon...
Done with H3K27ac and H3K4me1 intersection for sigmoid_colon.

The file for sigmoid_colon contains 14215 peaks.
-----------------------------------------------------
Processing stomach...
Intersecting ATAC-seq peaks with H3K27ac mark for stomach...
Done with H3K27ac intersection for stomach.
Intersecting ATAC-seq peaks with H3K4me1 mark for stomach...
Done with H3K4me1 intersection for stomach.
Intersecting H3K27ac and H3K4me1 peaks for stomach...
Done with H3K27ac and H3K4me1 intersection for stomach.

The file for stomach contains 8022 peaks.
-----------------------------------------------------
The total number of peaks in all files is 22237.
==================== All done! ====================</code></pre></details></li></ul></li></ol></details></li></ul><ul id="7f922ef6-67b1-43d3-869f-935f991d9d6a" class="toggle"><li><details open=""><summary>Focus on the regulatory elements that are located on chromosome 1 and generate a file <code>regulatory.elements.starts.tsv</code> that contains the name of the regulatory region and the start (5&#x27;) coordinate of the region:</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="632ef0a3-cb68-4dff-b865-bf9096742072" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all"># Define the list of tissues
tissues=(&quot;sigmoid_colon&quot; &quot;stomach&quot;)

# Loop over each tissue
for tissue in &quot;${tissues[@]}&quot;; do
    # Print a message to let the user know which tissue is being processed
    echo &quot;Processing $tissue...&quot;

    # Use awk to filter for lines where the first field is &quot;chr1&quot;, then print the fourth and second fields
    # Redirect the output to the appropriate .tsv file
    awk &#x27;BEGIN{FS=OFS=&quot;\t&quot;} $1==&quot;chr1&quot; {print $4, $2}&#x27; regulatory_elements/analyses/peaks.analyses/peaks.regulatory.H3K4me1.AND.H3K27ac.&quot;$tissue&quot;.bed &gt; regulatory_elements/data/tsv/regulatory.elements.starts.&quot;$tissue&quot;.tsv
done

echo &quot;All done!&quot;</code></pre></details></li></ul><ul id="1d415325-fa5f-4672-a445-1a35eb24b7d7" class="toggle"><li><details open=""><summary>Focus on protein-coding genes located on chromosome 1. From the BED file of the gene body coordinated that you generated, prepare a tab-separated file called gene.starts.tsv which will store the name of the gene in the first column, the start coordinate of the gene on the second column:</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="0bd896d7-d402-4934-9ba2-c7a4370b13db" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all"># Extract gene starts for protein-coding genes on chromosome 1
awk &#x27;BEGIN{FS=OFS=&quot;\t&quot;} $1==&quot;chr1&quot; {start = ($6==&quot;+&quot;) ? $2 : $3; print $4, start}&#x27; ../ChIP-seq/annotation/gencode.v24.protein.coding.gene.body.bed &gt; regulatory_elements/data/tsv/gene.starts.tsv

done</code></pre></details></li></ul><ul id="a16165fb-b31e-42ac-8e1d-6e9322e5e853" class="toggle"><li><details open=""><summary>Download or copy the python script inside <code>epigenomics_uvic/bin</code> folder. Have a look at the help page of the script to undertand how it works:</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="5146ff4b-1769-4d9c-8f71-ec53427df44c" class="code"><code class="language-Python" style="white-space:pre-wrap;word-break:break-all">#!/usr/bin/env python

#************
# LIBRARIES *
#************

import sys
from optparse import OptionParser

#*****************
# OPTION PARSING *
#*****************

parser = OptionParser()
parser.add_option(&quot;-i&quot;, &quot;--input&quot;, dest=&quot;input&quot;)
parser.add_option(&quot;-s&quot;, &quot;--start&quot;, dest=&quot;start&quot;)
options, args = parser.parse_args()

open_input = open(options.input)
enhancer_start = int(options.start)

#********
# BEGIN *
#********

x=1000000 # set maximum distance to 1 Mb
selectedGene=&quot;&quot; # initialize the gene as empty
selectedGeneStart=0 # initialize the start coordinate of the gene as empty

for line in open_input.readlines(): # for each line in the input file
    gene, y = line.strip().split(&#x27;\t&#x27;) # split the line into two columns based on a tab 
    position = int(y) # define a variable called position that correspond to the integer of the start of the gene
    diff = abs(position - enhancer_start) # compute the absolute value of the difference between position and enhancer_start

    if diff &lt; x: # if this absolute value is lower than x
        x = diff # this value will now be your current x
        selectedGene = gene # save gene as selectedGene
        selectedGeneStart = position # save position as selectedGeneStart

print &quot;\t&quot;.join([selectedGene, str(selectedGeneStart), str(x)])</code></pre><figure class="block-color-gray_background callout" style="white-space:pre-wrap;display:flex" id="2762647e-97d8-4ccb-bd5c-501622b00df7"><div style="font-size:1.5em"><span class="icon">💡</span></div><div style="width:100%">To make sure your script is working fine, run the following command:<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="5d906f3d-3d02-4b8c-8f87-760958a4f4c1" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all">python ../bin/get.distance.py --input regulatory_elements/data/tsv/gene.starts.tsv --start 98000</code></pre></div></figure></details></li></ul><ul id="59b47254-8ed4-476b-85ea-f6435c2d197c" class="toggle"><li><details open=""><summary>Finding the closest gene and the distance to the closest gene for each regulatory element contained in the file <code>regulatory.elements.tsv.</code></summary><figure class="block-color-gray_background callout" style="white-space:pre-wrap;display:flex" id="7f9c53c9-b9e2-41e9-8f48-e1fd006a6a29"><div style="font-size:1.5em"><span class="icon">💡</span></div><div style="width:100%">Using the python script <code><a href="http://get.distance.py/">get.distance.py</a></code>, it retrieves the closest gene and the distance to the closest gene for each regulatory element and for each tissue</div></figure><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="84dc8a29-88ed-4f2e-872a-e1ab6a59597c" class="code"><code class="language-Shell" style="white-space:pre-wrap;word-break:break-all"># Define the list of tissues
tissues=(&quot;sigmoid_colon&quot; &quot;stomach&quot;)

# Loop over each tissue
for tissue in &quot;${tissues[@]}&quot;; do
    echo &quot;Processing $tissue...&quot;

    # Find the closest gene and the distance to the closest gene for each regulatory element
    while read element start; do
        python ../bin/get.distance.py --input regulatory_elements/data/tsv/gene.starts.tsv --start $start
    done &lt; regulatory_elements/data/tsv/regulatory.elements.starts.&quot;$tissue&quot;.tsv &gt; regulatory_elements/data/tsv/regulatoryElements.genes.distances.&quot;$tissue&quot;.tsv

done</code></pre></details></li></ul><ul id="b59f0e48-bdd9-49ac-8822-503d1cd7d872" class="toggle"><li><details open=""><summary>Computing the mean and median of the distances to the closest gene for each tissue:</summary><script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js" integrity="sha512-7Z9J3l1+EYfeaPKcGXu3MS/7T+w19WtKQY/n+xzmw4hZhJ9tyYmcUS+4QqAlzhicE5LAfMQSF3iFTK9bQdTxXg==" crossorigin="anonymous" referrerPolicy="no-referrer"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" integrity="sha512-tN7Ec6zAFaVSG3TpNAKtk4DOHNpSwKHxxrsiw4GHKESGPs5njn/0sMCUMl2svV4wo4BK/rCP7juYz+zx+l6oeQ==" crossorigin="anonymous" referrerPolicy="no-referrer"/><pre id="7cc6e00e-1351-4837-a721-c4e867018a95" class="code"><code class="language-R" style="white-space:pre-wrap;word-break:break-all"># Load the data
regulatory.sigmoid_colon &lt;- read.table(file = &#x27;/epigenomics_uvic/ATAC-seq/regulatory_elements/data/tsv/regulatoryElements.genes.distances.sigmoid_colon.tsv&#x27;, sep = &#x27;\t&#x27;, header = FALSE)

regulatory.stomach &lt;- read.table(file = &#x27;/epigenomics_uvic/ATAC-seq/regulatory_elements/data/tsv/regulatoryElements.genes.distances.stomach.tsv&#x27;, sep = &#x27;\t&#x27;, header = FALSE)

# Compute the mean and median distances for each tissue
mean.sigmoid_colon &lt;- mean(regulatory.sigmoid_colon$V3)
median.sigmoid_colon &lt;- median(regulatory.sigmoid_colon$V3)

mean.stomach &lt;- mean(regulatory.stomach$V3)
median.stomach &lt;- median(regulatory.stomach$V3)

# Print the results
cat(&quot;Mean distance for sigmoid colon: &quot;, mean.sigmoid_colon)
cat(&quot;Median distance for sigmoid colon: &quot;, median.sigmoid_colon)

cat(&quot;Mean distance for stomach: &quot;, mean.stomach)
cat(&quot;Median distance for stomach: &quot;, median.stomach)</code></pre></details></li></ul><ul id="9b161ef3-d34e-43ef-88b0-25f83f84060e" class="toggle"><li><details open=""><summary>Results:</summary><blockquote id="0777c587-8663-4002-9bbe-3b910e6b4f95" class="">Print the results<p id="7bae8d94-cf71-49ac-89f3-cd88d67f7c87" class="">cat(&quot;Mean distance for sigmoid colon: &quot;, mean.sigmoid_colon)<br/>Mean distance for sigmoid colon:  73635.89<br/>cat(&quot;Median distance for sigmoid colon: &quot;, median.sigmoid_colon)<br/>Median distance for sigmoid colon:  35802<br/></p><p id="20d5e66b-fd50-460d-b5c2-2c1352e6b32a" class="">cat(&quot;Mean distance for stomach: &quot;, mean.stomach)<br/>Mean distance for stomach:  45227.05<br/>cat(&quot;Median distance for stomach: &quot;, median.stomach)<br/>Median distance for stomach:  27735<br/></p></blockquote><div id="cb15ba72-f7a1-4c55-bd18-13e9278d66c0" class="column-list"><div id="fbb17c31-3d98-4a2d-8be3-b233444a4d5c" style="width:16.666666666666664%" class="column"><p id="707a0bc7-909f-428c-b305-b7b09618a7ae" class=""><strong>Tissue</strong></p><p id="a2e11089-f270-419d-b6c9-68a906777363" class="">Sigmoid_colon</p><p id="b69bb1b1-8106-46cb-8650-3922a14fb8b4" class="">Stomach</p></div><div id="247aad1b-567a-4e4b-90e4-751cb6d97002" style="width:12.5%" class="column"><p id="7bb0327d-1c04-404c-a66b-ed504ae4e046" class=""><strong>Mean </strong></p><p id="a6602e55-ab10-4d8a-87d3-eea2f06cbdef" class="">73635.89      </p><p id="ea669bc9-5958-469a-a94c-4448ec3e8079" class="">45227.05</p><p id="8a93690c-eaef-406d-8d9d-2b58643ceffb" class="">
</p></div><div id="3466ecd1-9db9-45ca-9045-7626c8450737" style="width:187.5%" class="column"><p id="a4fbb1cd-83c1-4f6b-b164-37cf6835cea3" class=""><strong>Median </strong></p><p id="6319fdf9-800b-4ee7-890e-c9bbf42d5bf1" class="">35802 </p><p id="3db00adf-035e-403e-8b1f-7957b8edc157" class="">27735</p></div></div></details></li></ul></div></details></div></article><span class="sans" style="font-size:14px;padding-top:2em"></span></body></html>